source/slang/parameter-binding.cpp

```diff
@@ -999,7 +999,7 @@ static bool findLayoutArg(
     {
         if( modifier )
         {
-            *outVal = (UInt) strtoull(modifier->valToken.Content.Buffer(), nullptr, 10);
+            *outVal = (UInt) strtoull(String(modifier->valToken.Content).Buffer(), nullptr, 10);
```

Comment:
Just to confirm: the conversion to `String` here is motivated by the fact that we need a termination nul byte in order to call `strtoull` and `UnownedStringSlice` doesn't guarantee that, correct?

Alternatively, this should probably be calling out to a function to extract the value of an integer literal token, rather than calling `strtoull` directly.

Reply:
Yes - it's for null termination. Doing the conversion ourselves could avoid this, but doing correctly does require some care.

Reply:
String->integer is something I'm comfortable with us doing for ourselves just to make sure we are independent of any C runtime implementation differences. We may also eventually want a `BigInt` sort of type to make sure we can handle arbitrarily large literals and apply bounds checking after we know what concrete type we are expecting.

String->float (and vice versa) is something I'd like us to not do in general, although I think we currently still do our own value extraction today (issue #332 is tracking this, and links the Google double-conversion library we might want to use). String->float for hexadecimal floating-point literals is a special case that we might need/want to implement directly - its specifically the decimal case that is kind of a Bad Idea to write from scratch.